test/contrasts.jl

```diff
@@ -355,5 +355,30 @@
     @testset "Non-unique levels" begin
         @test_throws ArgumentError ContrastsMatrix(DummyCoding(), ["a", "a", "b"])
     end
+
+    @testset "other string types" begin
+        using CSV
+
+        using StatsModels: ContrastsMatrix
+        using DataAPI: levels
+
+        x = ["a", "b", "c", "a", "a", "b"]
+        io = IOBuffer()
+        tab = CSV.write(io, (; x=x, ))
+        seekstart(io)
+        x1 = Tables.columntable(CSV.File(io)).x
+        # this doesn't work w/ Julia < 1.3 because of WeakRefStrings compat
+        # x1 = WeakRefStrings.String1.(x)
+        x1_levs = levels(x1)
+
+        @test issetequal(x, x1)
+
+        c1 = ContrastsMatrix(DummyCoding(), x1_levs)
+        c = ContrastsMatrix(DummyCoding(levels=["a", "b", "c"]), x1_levs)
+        @test c == c1
+        @test eltype(c.levels) == eltype(c1.levels) != eltype(c.contrasts.levels)
```

Comment:
For older versions, this will be equality straight across. Can you do `@static if VERSION < v"1.3"` or the like?

Reply:
Totally agree.